README.md

```diff
@@ -128,87 +130,79 @@ Number matching is weak - the number has to appear
 “flattening” it, which is to say turning it 
 into a list of pathname/value pairs called Fields. Quamina 
-defines a `Flattener` interface type and provides a 
-JSON-specific implementation in the `FJ` type.
+defines a `Flattener` interface type and has a built-in
+`Flattener` for JSON.
 
 `Flattener` implementations in general will have
 internal state and thus not be thread-safe.
 
-The `MatchesForJSONEvent` API must create a new 
-`FJ` instance for each event so that it 
-can be thread-safe.  This works fine, but creating a 
-new `FJ` instance is expensive enough to slow the 
-rate at which events can be matched by 15% or so.
-
-For maximum performance in matching JSON events, 
-you should create your own `FJ` instance with the 
-`NewFJ(Matcher)` method. You can then use 
-`FJ.Flatten(event)` API to turn multiple successive
-JSON events into `Field` lists and pass them to 
-`Matcher`'s `MatchesForFields()` API, but `FJ` 
-includes a convenience method `FlattenAndMatch(event)` 
-which will call the `Matcher` for you.  As long as 
-each thread has its own `Flattener` instance, 
-everything will remain thread-safe.
-
-Also note that should you wish to process events 
+Note that should you wish to process events 
 in a format other than JSON, you can implement 
-the `Flattener` interface and use that to process 
-events in whatever format into Field lists.
+the `Flattener` interface yourself.
 
 ## APIs
-
 **Note**: In all the APIs below, field names and values in both
 Patterns and Events must be valid UTF-8.  Unescaped characters
 smaller than 0x1F (illegal per JSON), and bytes with value
 greater than 0XF4 (can't occur in correctly composed UTF-8)
 are rejected by the APIs.
-
+### Control APIs
 ```go
-type Matcher interface {
-	AddPattern(x X, pat string) error
-	MatchesForJSONEvent(event []byte) ([]X, error)
-	MatchesForFields(fields []Field) []X
-	DeletePattern(x X) error
-}
-```
-
-Above are the operations provided by a Matcher. Quamina
-includes an implementation called `CoreMatcher` which
-implements `Matcher`.  In a forthcoming release it will
-provider alternate implementations that offer extra
-features.
+func New(...Option) (*Quamina, error)
 
-```go
-func NewCoreMatcher() *Matcher
+func WithMediaType(mediaType string) Option
+func WithFlattener(f Flattener) Option
+func WithPatternDeletion(b bool) Option
+func WithPatternStorage(ps LivePatternsState) Option 
 ```
+For example:
+
 ```go
-func pruner.NewMatcher() *Matcher
+q, err := quamina.New(quamina.New(quamina.WithMediaType("application/json")))
 ```
-
-Create new Matchers, take no arguments. The difference
-is that the `pruner.NewMatcher` version supports the
-`DeletePattern()` API. Be careful: It occasionally
-rebuilds the Matcher in stop-the-world fashion, so if you
-delete lots of Patterns in a large Matcher you may 
-encounter occasional elevated latencies.
+The meanings of the `Option` functions are:
+
+`WithMediaType`: In the futue, Quamina will support 
```

Comment:
question: do we need that option/make it part of the API? I thought picking/implementing the corresponding `Flattener` would implicitly do that?

Reply:
Haha, you're talking to an IETF guy.  The correct way to identify byte blobs crossing the Internet is with media types, so Flatteners should be keyed by those.  If you want to bring your own Flattener to support your private secret format, we support that.  But for something you got from an HTTP server, this is a good approach.

Reply:
I knew you came from that direction :)

There's just a problem here with the options then. Will explain in the corresponding code section. IMHO `Flatteners` should be keyed by mediatype, e.g. we own the defaultMatcher which is keyed by JSON. If someone wants to overwrite, they likely know the mediatype - hence my question why have two APIs basically doing the same thing?